To determine which features have a significant effect on depression, a statistical analysis will be conducted to identify the most influential factors. Depression is a complex mental health condition, and it is influenced by various biological, psychological, and social factors. By applying statistical techniques, we can better understand the relationships between these features and depression outcomes.

In this analysis, we aim to explore the dataset and apply statistical measures such as correlation analysis, hypothesis testing, and regression modeling. These methods will help us assess the strength and significance of the relationships between depression and various features, such as age, lifestyle habits, social factors, and more.

The insights gained from this analysis can contribute to building predictive models for depression, allowing for early identification of high-risk individuals and informing targeted interventions.

In [1]:
import pandas as pd
import numpy as np
df= pd.read_csv("clean_data.csv")

### 1. **Chi-Square Test (For Categorical Features)**
   - **Purpose**: To test the independence between categorical features and the target variable (depression status).
   - **How it Works**: The Chi-Square test checks if the distribution of the target variable is significantly different across the categories of each feature.
   - **Interpretation**: A low p-value (typically < 0.05) suggests that there is a significant association between the feature and the target variable.
 

In [6]:
from scipy.stats import chi2_contingency
categorical_columns = ["country", "Education", "Employed", "Disability", "Mental Health issue", "social media", 
                       "Hour spend on fitness", "wind down", "Mouth dryness","positive feeling"
              ,"breath difficulties", "difficulties in initiate things", "trembling", 
             "worrying situation", "feeling of nothing to look forward to", "down hearted",
             "enthusaistic about anything", "feel meaningless life", "scared"]
for col in categorical_columns:
    contingency_table = pd.crosstab(df[col], df['depression status'])
    chi2, p, dof, expected = chi2_contingency(contingency_table)
    print(f'{col}: p-value = {round(p, 5)}')

country: p-value = 1.0
Education: p-value = 1.0
Employed: p-value = 0.88346
Disability: p-value = 0.24548
Mental Health issue: p-value = 0.0
social media: p-value = 0.08767
Hour spend on fitness: p-value = 0.04616
wind down: p-value = 0.0
Mouth dryness: p-value = 0.00053
positive feeling: p-value = 0.0
breath difficulties: p-value = 0.0
difficulties in initiate things: p-value = 0.0
trembling: p-value = 0.0
worrying situation: p-value = 0.0
feeling of nothing to look forward to: p-value = 0.0
down hearted: p-value = 0.0
enthusaistic about anything: p-value = 0.0
feel meaningless life: p-value = 0.0
scared: p-value = 0.0


Here’s a summary of the Chi-Square test results for each column against the **depression status**, based on the p-values provided:

### **No Significant Association (p-value > 0.05)**:
1. **Country**: p-value = 1.0  
   - There is **no association** between a person's country and their depression status.
  
2. **Education**: p-value = 1.0  
   - There is **no association** between education level and depression status.
  
3. **Employed**: p-value = 0.88346  
   - Employment status is **not significantly related** to depression status.
  
4. **Disability**: p-value = 0.24548  
   - There is **no significant relationship** between having a disability and depression status.
  
5. **Social media usage**: p-value = 0.08767  
   - Social media usage **shows no significant association** with depression status, though it’s somewhat close to significance.
  
### **Significant Association (p-value ≤ 0.05)**:
1. **Mental health issue**: p-value = 0.0  
   - There is a **significant association** between having a mental health issue and depression status.

2. **Hours spent on fitness**: p-value = 0.04616  
   - The amount of time spent on fitness is **significantly associated** with depression status.
  
3. **Wind down**: p-value = 0.0  
   - The way a person winds down is **significantly related** to depression status.
  
4. **Mouth dryness**: p-value = 0.00053  
   - Experiencing mouth dryness has a **significant relationship** with depression status.
  
5. **Positive feeling**: p-value = 0.0  
   - A person's positive feelings are **significantly associated** with depression status.

6. **Breathing difficulties**: p-value = 0.0  
   - Difficulty in breathing is **significantly associated** with depression status.

7. **Difficulties in initiating things**: p-value = 0.0  
   - There is a **significant relationship** between struggling to initiate tasks and depression status.
  
8. **Trembling**: p-value = 0.0  
   - Trembling has a **strong significant association** with depression status.
  
9. **Worrying situations**: p-value = 0.0  
   - Worrying frequently is **significantly related** to depression status.
  
10. **Feeling of having nothing to look forward to**: p-value = 0.0  
    - A feeling of having nothing to look forward to is **strongly associated** with depression status.
  
11. **Feeling downhearted**: p-value = 0.0  
    - Being downhearted shows a **significant relationship** with depression status.
  
12. **Enthusiasm about anything**: p-value = 0.0  
    - Lack of enthusiasm is **significantly associated** with depression status.
  
13. **Feeling life is meaningless**: p-value = 0.0  
    - A sense that life is meaningless has a **significant association** with depression status.
  
14. **Feeling scared**: p-value = 0.0  
    - Being scared is **significantly related** to depression status.

### **Summary**:
- Variables such as **mental health issues, mouth dryness, breathing difficulties, difficulties in initiating tasks**, and emotional states (like feeling scared or downhearted) have a strong association with depression status.
- In contrast, factors like **country, education, and employment status** show no significant relationship with depression status.


### 2. **ANOVA (For Numerical Features)**
   - **Purpose**: To compare the means of a numerical feature across the different categories of the target variable (depression status).
   - **How it Works**: ANOVA tests whether the means of the numerical feature are significantly different across the different levels of the categorical target.
   - **Interpretation**: A low p-value indicates that there is a significant difference in the means of the feature across the different depression statuses.

In [8]:
from scipy.stats import f_oneway
numerical_columns = ["Study", "Age"]

for col in numerical_columns:
    groups = [df[col][df['depression status'] == status] for status in df['depression status'].unique()]
    f_stat, p = f_oneway(*groups)
    print(f'{col}: p-value = {p}')

Study: p-value = 0.5992298284316735
Age: p-value = 0.7389712913153184


The results from the ANOVA test indicate the following:

1. **Study**: p-value = 0.5992  
   - There is **no significant difference** in depression status based on the field of study. The p-value is much higher than 0.05, meaning the study field does not have a noticeable impact on depression status.

2. **Age**: p-value = 0.7389  
   - Age also shows **no significant association** with depression status. The p-value is greater than 0.05, suggesting that depression levels are not significantly different across age groups.

### **Summary**:
Both **study field** and **age** do not show significant differences or associations with depression status based on the ANOVA test results. This implies that neither factor plays a crucial role in determining depression status in this context.

### 3. **Logistic Regression (For Mixed Data Types)**
   - **Purpose**: To model the relationship between one or more features and a binary or ordinal target variable.
   - **How it Works**: Logistic regression provides coefficients that indicate the strength and direction of the relationship between each feature and the target variable.
   - **Interpretation**: Look at the p-values of the coefficients; low p-values suggest that the corresponding feature is a significant predictor of depression status.



In [14]:
X

,const,Study,Age
0,1.0,10,29
1,1.0,10,28
2,1.0,20,28
3,1.0,6,26
4,1.0,2,33
...,...,...,...
238,1.0,10,28
239,1.0,20,28
240,1.0,20,25
241,1.0,35,29


In [ ]:
import statsmodels.api as sm

X = df.copy()  # Select your features
y = X.pop('depression status')
X = sm.add_constant(df[numerical_columns])
model = sm.Logit(y, X).fit()
print(model.summary())

### 4. **Feature Importance with Tree-Based Models (e.g., Random Forest)**
   - **Purpose**: To rank the importance of different features in predicting the target variable.
   - **How it Works**: Tree-based models like Random Forests can measure feature importance by how much each feature reduces impurity (e.g., Gini impurity) across all trees in the forest.
   - **Interpretation**: Higher values indicate more important features.

In [ ]:
from sklearn.ensemble import RandomForestClassifier
X = df.copy()
y = X.pop('depression status')
rf = RandomForestClassifier()
rf.fit(X, df['depression status'])
feature_importances = pd.Series(rf.feature_importances_, index=X.columns).sort_values(ascending=False)
print(feature_importances)

### 5. **Correlation Matrix (For Numerical Features)**
   - **Purpose**: To examine the linear relationships between numerical features and the target variable.
   - **How it Works**: Calculate the correlation coefficient between each numerical feature and the target variable.
   - **Interpretation**: Values closer to +1 or -1 indicate stronger correlations.


In [ ]:
correlation_matrix = df.corr()
   print(correlation_matrix['depression status'].sort_values(ascending=False))


### 6. **Mutual Information (For Both Categorical and Numerical Features)**
   - **Purpose**: To measure the dependency between the feature and the target variable.
   - **How it Works**: Mutual Information quantifies the amount of information obtained about one variable through the other.
   - **Implementation**: Use `sklearn.feature_selection.mutual_info_classif` for classification tasks.
   - **Interpretation**: Higher scores indicate greater influence of the feature on the target variable.

In [ ]:
from sklearn.feature_selection import mutual_info_classif
X = df.copy()
y = X.pop('depression status')
mi_scores = mutual_info_classif(X, y)
mi_series = pd.Series(mi_scores, index=features).sort_values(ascending=False)
print(mi_series)


### 7. **Principal Component Analysis (PCA)**
- **Purpose**: To reduce the dimensionality of your dataset while retaining most of the variance.
- **How it Works**: PCA transforms your features into a new set of uncorrelated components that are linear combinations of the original features. The first few components typically explain most of the variance.
- **Interpretation**: Components with higher variance ratios are more important.


In [ ]:
from sklearn.decomposition import PCA
X = df.copy()

pca = PCA()
pca.fit(X)
print(pca.explained_variance_ratio_)